remotecfg: wire calls to RegisterCollector (#1359)
grafana/alloy: docs/sources/reference/config-blocks/remotecfg.md

`id`             | `string`             | A self-reported ID.                               | `see below` | no
`attributes`     | `map(string)`        | A set of self-reported attributes.                | `{}`        | no
`poll_frequency` | `duration`           | How often to poll the API for new configuration.  | `"1m"`      | no
`name`           | `string`             | A human-readable name for the collector.          | `""`        | no

If the `url` is not set, then the service block is a no-op.

> If not set, the self-reported `id` that {{< param "PRODUCT_NAME" >}} uses is a randomly generated, anonymous unique ID (UUID) that is stored as an `alloy_seed.json` file in {{< param "PRODUCT_NAME" >}}'s storage path so that it can persist across restarts. In that case, the `name` field can be used to set another human-friendly identifier for the specific Alloy instance.

Reviewer: ```suggestion
If not set, the self-reported `id` that {{< param "PRODUCT_NAME" >}} uses is a randomly generated, anonymous unique ID (UUID) that is stored as an `alloy_seed.json` file in the {{< param "PRODUCT_NAME" >}} storage path so that it can persist across restarts.
You can use the `name` field to set another human-friendly identifier for the specific {{< param "PRODUCT_NAME" >}} instance.
```
